apps/docs/content/guides/database/extensions/pgaudit.mdx

```diff
@@ -41,68 +46,389 @@ To disable an extension you can call `drop extension`.
+
+-- check to make sure the changes are finalized:
+SELECT
+  rolname,
+  rolconfig
+FROM pg_roles
+WHERE rolname = 'zapier';
+```
+
+### Global Logging
 
+<Admonition type="caution">
+  Use global logging sparingly as it can generate millions of logs per hour. Consider alternatives
+  first.
 </Admonition>
 
-## Example
+The below SQL configures pgaudit to record all events associated with the "postgres" role. Since postgres has extensive privileges, this effectively monitors all database activity.
 
-Given a `pgaudit` setting
+```sql
+ALTER ROLE postgres SET pgaudit.log TO 'all';
+```
+
+To check if the postgres role is auditing, execute the following command:
 
-{/* prettier-ignore */}
 ```sql
-set pgaudit.log = 'read, ddl';
+select
+  rolname,
+  rolconfig
+from pg_roles
+where rolname = 'postgres';
+-- should return a rolconfig path with "pgaudit.log=all" present
 ```
 
-The following create table, insert and select statements
+To remove the settings, execute the following code:
+
+```
+ALTER ROLE postgres SET pgaudit.log TO 'none';
+```
+
+## Object Mode Settings
+
+With object mode, you assign database users to the `pgaudit.role` variable. Whatever the users could hypothetically do, pgaudit will log.
+
+For re-emphasis: when you assign a role to pgaudit, it logs any actions that role is _capable_ of performing, regardless of who or what actually initiated the event.
+
+### Object Logging
+
+To fine-tune what pgaudit will record, you must create a custom database role with limited permissions:
+
+```sql
+-- can be named anything
+CREATE ROLE some_audit_role NOINHERIT;
+```
+
+The `ROLE` cannot be impersonated by any other user. It cannot be logged into or used. In this case, it solely exists to securely define what pgaudit will record.
+
+Once the role is created, you can direct pgaudit to log by assigning it to the `pgaudit.role` variable:
+
+```sql
+ALTER ROLE postgres SET pgaudit.role TO 'some_audit_role';
+```
+
+You can then assign the role to monitor only approved events, such as `SELECT` statements that include a specific table:
+
+```sql
+GRANT SELECT ON random_table TO some_audit_role;
+```
+
+With this privilege granted, pgaudit will record all select statements against the `random_table`. Note, that this does include joins that feature the table, too. All privileges can be viewed in the [PostgreSQL documentation](https://www.postgresql.org/docs/current/ddl-priv.html).
+
+If you would no longer like to use object logging, you need to modify the pgaudit.role variable:
+
+```sql
+-- change pgaudit.role to no longer reference some_audit_role
+ALTER ROLE postgres SET pgaudit.role to '';
+
+-- view if pgaudit.role changed with the following command:
+SELECT
+  rolname,
+  rolconfig
+FROM pg_roles
+WHERE rolname = 'postgres';
+```
+
+If you are no longer using a `ROLE`, it is best to [`DROP` it](https://stackoverflow.com/questions/51256454/cannot-drop-postgresql-role-error-cannot-be-dropped-because-some-objects-depe):
```

Comment:
Supabase's [role's docs](https://supabase.com/docs/guides/database/postgres/roles) doesn't provide instructions on how to `DROP` custom users. I suppose it would be better to add to the docs than link externally